### ISSUE TITLE
URL, inserted in chat, becomes incorrect, due to a Message Styling

### ISSUE THREAD

**Issue by DX37** [qTox/qTox#4233]
OS: I think - All
qTox version: v1.8.1-98-ge65e45aa
Commit hash: e65e45aa
toxcore: 0.1.6
Qt: 5.8.0

Reproducible: Always

##### Steps to reproduce
Just insert URL in chat and send.

##### Observed Behavior
![_2017-03-03_21-57-10](https://cloud.githubusercontent.com/assets/12829428/23555687/7565d8ac-005c-11e7-89d2-083a4bab61d4.png)
URL, which opens in browser, looks like this: `https://<i>github.com<i>qtox</i>qTox<i>pull</i>4232`

##### Expected Behavior
Normal URL?

##### Additional Info
upd: When you clicking at URL, it opens incorrectly. When you open it after its copying, it opens fine.

And I freaking know why it become looking like that!
https://github.com/qTox/qTox/blob/master/doc/user_manual_en.md#message-styling
![_2017-03-03_22-00-10](https://cloud.githubusercontent.com/assets/12829428/23555800/ddd19f98-005c-11e7-8d63-3219115f3a6f.png)

**Comment by zetok** [comment]
A regression, v1.8.1 is not affected.

`git bisect` in range v1.8.0...e65e45aafe06b02c100474d13245370a4aba4257:

5047a65e119ac5094f67775c7ef18dd0e7eee9bc is the first bad commit
```
commit 5047a65e119ac5094f67775c7ef18dd0e7eee9bc
Author: noavarice <alex9razinkov@gmail.com>
Date:   Mon Feb 6 18:20:45 2017 +0300

    refactor: improved HTML-formatting mechanism for text messages
    
    Added class provides text font formatting with HTML font tags
    Supported stuff:
      - nested formatting;
      - several text pieces formatted with the same formatting style at the
        one message;
      - styling applies only if non-whitespace symbol follows opening
        formatting symbol and also non-whitespace symbol preceds closing
        symbol;
      - only multiline code font formatting supports new line inside of
        message text.
    Also fix #3804
```

cc @noavarice

**Comment by noavarice** [comment]
I observed this problem a bit earlier than that issue was posted so I'm about to fix this
